cmds/core/ls/ls_linux_test.go

```diff
@@ -23,26 +25,35 @@ import (
 //     = mkdev(0x456, 0x12378)
 //     = (0x12378 & 0xff) | (0x456 << 8) | ((0x12378 & ~0xff) << 12)
 //     = 0x12345678
-func TestLargeDevNumber(t *testing.T) {
-	if uid := os.Getuid(); uid != 0 {
-		t.Skipf("test requires root, your uid is %d", uid)
-	}
 
-	// Make the node.
-	tmpDir := t.TempDir()
-	file := filepath.Join(tmpDir, "large_node")
-	if err := unix.Mknod(file, 0o660|unix.S_IFBLK, 0x12345678); err != nil {
-		t.Fatal(err)
+// Test listName func
+func TestListNameLinux(t *testing.T) {
+	testutil.SkipIfNotRoot(t)
+	// Create a directory
+	d := t.TempDir()
+	if err := unix.Mknod(filepath.Join(d, "large_node"), 0o660|unix.S_IFBLK, 0x12345678); err != nil {
```

Comment:
It seems to me the defer should be right after t.TempDir, and the Error for the Mknod failing should be Fatal? This is going to be a  very confusing failure otherwise.